### PR TITLE
Segmented algorithms module

### PR DESCRIPTION
**Pull request by msimberg**
Same deal as #3957 (also depends on it), but for segmented algorithms.

### CODE REVIEW

**Comment by hkaiser** [comment]
The following file should be added to this module:
```
#include <hpx/traits/segmented_iterator_traits.hpp> 
```

**Comment by msimberg** [comment]
> The following file should be added to this module:
> 
> ```
> #include <hpx/traits/segmented_iterator_traits.hpp> 
> ```

Thanks, good catch! I'll update this first thing on Monday.

File: libs/algorithms/CMakeLists.txt [hunk 1]
```diff
@@ -124,6 +124,6 @@ add_hpx_module(algorithms
     # TODO: Really depends on main HPX library, but cannot add that as a
     # dependency yet. Need to have the parts that algorithms depends on in
     # separate modules first.
-    DEPENDENCIES hpx_assertion hpx_config hpx_timing
+    DEPENDENCIES hpx_assertion hpx_config hpx_iterator_support hpx_timing
```
Comment (hkaiser) [inline libs/algorithms/CMakeLists.txt]: Wouldn't this create a circular dependency between the `algorithms` and `segmented_algorithm` modules?

Comment (msimberg) [reply libs/algorithms/CMakeLists.txt]: Should not as far as I can tell, but I may have missed something. What is the cycle you're thinking about? FYI, @aurianer is working on getting the checks for cyclic dependencies. I'll have a look in the morning to make sure there's no cycle.

Comment (hkaiser) [reply libs/algorithms/CMakeLists.txt]: Hmmm, not sure what I was thinking. There is no cycle for sure - my brain was trying to convince me that there was one, however ;-) Sorry for the noise.

**Pull request merged**